query/src/sessions/session_mgr.rs

```diff
@@ -187,7 +191,7 @@ impl SessionManager {
 
     #[allow(clippy::ptr_arg)]
     pub fn get_session_by_id(self: &Arc<Self>, id: &str) -> Option<SessionRef> {
-        let sessions = self.active_sessions.read();
+        let sessions = futures::executor::block_on(self.active_sessions.read());
```

Comment:
I am worried about the `active_sessions` lock here, it's a async and we `block_on` it to sync here, it seems not a good way.

Reply:
I initially wanted to unify into all to async approach, but when I want to make [processes_info](https://github.com/datafuselabs/databend/pull/4361/files#diff-05e190acf915432f50b78d25482d816df331623745159f6e235f255aa4c73516R247) to async, I found it will make a huge change to the codebase...
The [ProcessesTable](https://github.com/datafuselabs/databend/blob/51b0ba2667e84e911c6b6309803713de0a1cd0f1/query/src/storages/system/processes_table.rs#L45) implement the `SyncSystemTable`, and `SyncSystemTable` is a sync trait and implemented by many tables. So I choose the block_one way.

But for this `get_session_by_id`, change it to the async function is easy.

Reply:
So let's first make CI pass, then I'll try to change all related block_on to async way.

Reply:
We can make `ProcessesTable` to AsyncSystemTable using the new processor.
@zhang2014  maybe give some suggestions here

Reply:
:cry:

Reply:
if the outter async runtime is tokio, `futures::executor::block_on` might causes stuck as https://github.com/tokio-rs/tokio/issues/2603 describes.